LocalFolder/explainer.md

```diff
@@ -0,0 +1,129 @@
+    if (opfsRoot) {
+        await opfsRoot.removeEntry("microsoft_store_app_local_folder_APPID.37853FC22B2CE_6rarf9sa4v8jt", { recursive: true }); 
+    }
+```
+
+Alternatively, contents in LocalFolder can be deleted by calling `remove()` on file and directory handles directory.
+
+```JS
+    let localStorage = await opfsRoot.getDirectoryHandle("microsoft_store_app_local_folder_APPID.37853FC22B2CE_6rarf9sa4v8jt", {create: false}); 
+
+    for await (const [key, value] of localStorage.entries()) {
+        value.remove();
+    }
+```
+
+### Entry name
+
+The LocalFolder entry within the OPFS root directory can be found under the name `microsoft_store_app_local_folder{PFN}` where `{PFN}` is the Package Family Name of the app.
+
+### Microsoft Store PWA context
+
+To be in a Microsoft Store app context, `navigator.storage.getDirectory()` must be called from:
+* Within a top level document
+* Within a web app window of a Microsoft Store installed PWA
+* From the main thread and not a Web Worker
+
+### Avoiding collisions
+To avoid collisions, the LocalFolder entry under the OPFS root directory is assigned a specific name that contains the Package Family Name. If an entry with the same name already exists under the OPFS root directory, an entry representing the LocalFolder directory will not be made visible until the existing entry is renamed or otherwise removed.
+
+## Similar storage folders
+
+There are other UWP storage APIs similar to LocalFolder that can be exposed the same way through OPFS directory handles. 
+
+* [LocalCacheFolder](https://learn.microsoft.com/en-us/uwp/api/windows.storage.applicationdata.localfolder?view=winrt-22621)
+* [RoamingFolder](https://learn.microsoft.com/en-us/uwp/api/windows.storage.applicationdata.roamingfolder?view=winrt-22621)
+* [SharedLocalFolder](https://learn.microsoft.com/en-us/uwp/api/windows.storage.applicationdata.sharedlocalfolder?view=winrt-22621)
+* [TemporaryFolder](https://learn.microsoft.com/en-us/uwp/api/windows.storage.applicationdata.temporaryfolder?view=winrt-22621)
+
+## Alternatives considered
+
+### New Web API
+
+We considered creating a new API on `navigator` to return a `FileSystemDirectoryHandle` of the app's LocalFolder directory. This was rejected primarily because of the long term cost of maintaining a non-standard API. 
+
+### File migration
+
+We considered a solution that migrates the contents of the LocalFolder directory into OPFS storage. The drawbacks to this solution includes:
+
+* It could take a long time to migrate large files. There is no way for the calling code to register a callback to wait for migration to be completed without introducing a new API, negating the advantage of not creating a new API on `navigator`.
+* Migration of files could fail due the lack of storage space and other I/O errors. There is no good way to communicate such errors to the calling code without introducing a new API.
+
+## Privacy And Security Considerations
```

Comment:
Just in terms of fully documenting security impact, does the [storage access permissions policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Permissions-Policy/storage-access) apply to OPFS at all, in terms of being able to limit access to iframes?

Reply:
Storage Access API (and the `storage-access` permission policy seems to only control access to cookie storage (window.cookie). It does not seem relevant to OPFS.